src/resolve.ts

```diff
@@ -59,11 +76,141 @@ async function loadCache(cacheFilePath: string): Promise<CacheSchemaV3> {
+  if (// See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/link-resolver.js#L14
+      packagePattern.startsWith('link:') ||
+      // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/file-resolver.js#L18
+      packagePattern.startsWith('file:') ||
+      /^\.{1,2}\//.test(packagePattern) ||
+      pathLib.isAbsolute(packagePattern) ||
+      // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/tarball-resolver.js#L15
+      packagePattern.startsWith('http://') ||
+      packagePattern.startsWith('https://') ||
+      (packagePattern.indexOf('@') < 0 && (packagePattern.endsWith('.tgz') || packagePattern.endsWith('.tar.gz'))) ||
+      // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/github-resolver.js#L6
+      packagePattern.startsWith('github:') ||
+      /^[^:@%/\s.-][^:@%/\s]*[/][^:@\s/%]+(?:#.*)?$/.test(packagePattern) ||
+      // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/gitlab-resolver.js#L6
+      packagePattern.startsWith('gitlab:') ||
+      // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/bitbucket-resolver.js#L6
+      packagePattern.startsWith('bitbucket:') ||
+      // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/gist-resolver.js#L26
+      packagePattern.startsWith('gist:') ||
+      // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/git-resolver.js#L19
+      /^git:|^git\+.+:|^ssh:|^https?:.+\.git$|^https?:.+\.git#.+/.test(packagePattern)) {
+    return true
+  } else {
+    // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/git-resolver.js#L19
+    const {hostname, path} = url.parse(packagePattern);
+    if (hostname && path && ['github.com', 'gitlab.com', 'bitbucket.com', 'bitbucket.org'].indexOf(hostname) >= 0) {
+      return path.split('/').filter((p): boolean => !!p).length === 2;
+    }
+  }
+}
+
+function resolvePackageNameFromRegistryYarnPackagePattern(packagePattern: string): string {
+  // See https://github.com/yarnpkg/yarn/blob/953c8b6a20e360b097625d64189e6e56ed813e0f/src/resolvers/exotics/registry-resolver.js#L12
+  const match = packagePattern.match(/^(\S+):(@?.*?)(@(.*?)|)$/);
+  if (match) {
+    return match[2];
+  } else {
+    throw new Error(`Failed to resolve yarn package pattern ${packagePattern}, unrecognized format`)
+  }
+}
+
+function resolveNpmPackagesFromYarnLockDependencies(yarnLockDependencies: YarnLockDependency[]): PackageWithId[] {
+  const packages: PackageWithId[] = yarnLockDependencies
+    .reduce(function(filterMappedDependencies: PackageWithId[], {packagePattern, version}) {
```

Comment:
Use arrow function syntax, please :)

Reply:
9ce1e74b80130cbf977254416e052b39e4b867a7
Added lint rule as well which actually was able to autofix this case.